### DISCUSSION TITLE
How to ignore Duplicate Rule when commitWork for Account/Contact/Lead?

### DISCUSSION THREAD

**Discussion by briancodey** [apex-enterprise-patterns/fflib-apex-common#445] [Q&A] [answered]
Hi,

Salesforce implemented Duplicate Rules for Account/Contact/Lead.  Database.DMLOptions allows us to ignore the duplicate rule and force insert into the db.

Database.DMLOptions dml = new Database.DMLOptions(); 
dml.DuplicateRuleHeader.allowSave = true;
Database.insert(con, dml);

When using Unit Of Work, how do I achieve the same?

Thanks,
Brian

**Comment by ImJohnMDaniel** [comment, marked as answer]
G'day @briancodey.  When using the UnitOfWork, the [default constructor](https://github.com/apex-enterprise-patterns/fflib-apex-common/blob/master/sfdx-source/apex-common/main/classes/fflib_SObjectUnitOfWork.cls#L157) creates the instance of the inner class [`SimpleDML`](https://github.com/apex-enterprise-patterns/fflib-apex-common/blob/master/sfdx-source/apex-common/main/classes/fflib_SObjectUnitOfWork.cls#L96).  The `fflib_SObjectUnitOfWork` class relies on the [`IDML` interface](https://github.com/apex-enterprise-patterns/fflib-apex-common/blob/master/sfdx-source/apex-common/main/classes/fflib_SObjectUnitOfWork.cls#L87) to dictate how DML operations will be handled.  

With that in mind, you would create your own `IDML` class that sets the `DMLOptions` the way you need them, and then instantiate the `fflib_SObjectUnitOfWork` using its [second constructor](https://github.com/apex-enterprise-patterns/fflib-apex-common/blob/master/sfdx-source/apex-common/main/classes/fflib_SObjectUnitOfWork.cls#L160) which accepts `IDML` as the second argument.  For an alternate example of an `IDML`, refer to the other one present in the fflib_SObjectUnitOfWork class called [`UserModeDML`](https://github.com/apex-enterprise-patterns/fflib-apex-common/blob/master/sfdx-source/apex-common/main/classes/fflib_SObjectUnitOfWork.cls#L125).

I hope this helps.  Let us know if you have more questions.

**Reply by briancodey** [reply]
Thanks, @ImJohnMDaniel.
